app/controllers/profile/orders_controller.rb

```diff
@@ -29,6 +29,16 @@ def create
     end
   end
 
+  def update
+    order = Order.find(params[:id])
+    unless order.cancelled? || order.shipped?
+      order.cancel
```

Comment:
I like that you used a model method here, and the naming convention makes it clear what it does.